src/Persistence/Mapping/Driver/PHPDriver.php

```diff
@@ -35,6 +39,33 @@ public function loadMetadataForClass(string $className, ClassMetadata $metadata)
      */
     protected function loadMappingFile(string $file): array
     {
+        try {
+            $callback = Closure::bind(static function ($file): mixed {
+                $metadata = null;
+
+                return include $file;
+            }, null, null)($file);
+        } catch (CompileError $e) {
+            throw $e;
+        } catch (Error) {
+            // Accessing the undefined $metadata variable or $this causes an Error
```

Comment:
I don't get it… when I modify this catch statement to retry the exception, here is what is printed:

```
Error: Call to a member function getFieldNames() on null
```

There is no mention of accessing a `$metadata` property on $this.